homeassistant/components/doods/image_processing.py

```diff
@@ -0,0 +1,366 @@
+        vol.Optional(CONF_TOP, default=0): cv.small_float,
+    }
+)
+
+LABEL_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): cv.string,
+        vol.Optional(CONF_AREA): AREA_SCHEMA,
+        vol.Optional(CONF_CONFIDENCE): vol.Range(min=0, max=100),
+    }
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_URL): cv.string,
+        vol.Required(CONF_DETECTOR): cv.string,
+        vol.Optional(CONF_AUTH_KEY, default=""): cv.string,
+        vol.Optional(CONF_FILE_OUT, default=[]): vol.All(cv.ensure_list, [cv.template]),
+        vol.Optional(CONF_CONFIDENCE, default=0.0): vol.Range(min=0, max=100),
+        vol.Optional(CONF_LABELS, default=[]): vol.All(
+            cv.ensure_list, [vol.Any(cv.string, LABEL_SCHEMA)]
+        ),
+        vol.Optional(CONF_AREA): AREA_SCHEMA,
+    }
+)
+
+
+def draw_box(draw, box, img_width, img_height, text="", color=(255, 255, 0)):
+    """Draw bounding box on image."""
+    ymin, xmin, ymax, xmax = box
+    (left, right, top, bottom) = (
+        xmin * img_width,
+        xmax * img_width,
+        ymin * img_height,
+        ymax * img_height,
+    )
+    draw.line(
+        [(left, top), (left, bottom), (right, bottom), (right, top), (left, top)],
+        width=5,
+        fill=color,
+    )
+    if text:
+        draw.text((left, abs(top - 15)), text, fill=color)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Doods client."""
+    url = config[CONF_URL]
+    auth_key = config[CONF_AUTH_KEY]
+    detector_name = config[CONF_DETECTOR]
+
+    doods = PyDOODS(url, auth_key)
+    response = doods.get_detectors()
+    if not isinstance(response, dict):
+        _LOGGER.warning("Could not connect to doods server: %s", url)
+        return
+
+    detector = {}
+    for server_detector in response["detectors"]:
+        if server_detector["name"] == detector_name:
+            detector = server_detector
+            break
+
+    if not detector:
+        _LOGGER.warning(
+            "Detector %s is not supported by doods server %s", detector_name, url
+        )
+        return
+
+    entities = []
+    for camera in config[CONF_SOURCE]:
+        entities.append(
+            Doods(
+                hass,
+                camera[CONF_ENTITY_ID],
+                camera.get(CONF_NAME),
+                doods,
+                detector,
+                config,
+            )
+        )
+    add_entities(entities)
+
+
+class Doods(ImageProcessingEntity):
+    """Doods image processing service client."""
+
+    def __init__(self, hass, camera_entity, name, doods, detector, config):
+        """Initialize the DOODS entity."""
+        self.hass = hass
+        self._camera_entity = camera_entity
+        if name:
+            self._name = name
+        else:
+            self._name = "Doods {0}".format(split_entity_id(camera_entity)[1])
+        self._doods = doods
+        self._file_out = config.get(CONF_FILE_OUT)
```

Comment:
Please use `dict[key]` for required config keys and keys with default config schema values.

Reply:
@MartinHjelmare I'm not sure I understand what you mean.

Reply:
We should only use `dict.get` if we don't know that the key is in the dict. If we know that the key is in the dict, like we do here, we do `dict[key]`.
```suggestion
        self._file_out = config[CONF_FILE_OUT]
```

Reply:
I think I got them all.